src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,485 @@
+		'wp_privacy_data_removal_requests_list_table' => 'wp-admin/includes/class-wp-privacy-data-removal-requests-list-table.php',
+		'wp_privacy_policy_content'                   => 'wp-admin/includes/class-wp-privacy-policy-content.php',
+		'wp_privacy_requests_table'                   => 'wp-admin/includes/class-wp-privacy-requests-table.php',
+		'wp_screen'                                   => 'wp-admin/includes/class-wp-screen.php',
+		'wp_site_health_auto_updates'                 => 'wp-admin/includes/class-wp-site-health-auto-updates.php',
+		'wp_site_health'                              => 'wp-admin/includes/class-wp-site-health.php',
+		'wp_site_icon'                                => 'wp-admin/includes/class-wp-site-icon.php',
+		'wp_terms_list_table'                         => 'wp-admin/includes/class-wp-terms-list-table.php',
+		'wp_theme_install_list_table'                 => 'wp-admin/includes/class-wp-theme-install-list-table.php',
+		'wp_themes_list_table'                        => 'wp-admin/includes/class-wp-themes-list-table.php',
+		'wp_upgrader_skin'                            => 'wp-admin/includes/class-wp-upgrader-skin.php',
+		'wp_upgrader'                                 => 'wp-admin/includes/class-wp-upgrader.php',
+		'wp_users_list_table'                         => 'wp-admin/includes/class-wp-users-list-table.php',
+
+		/* Classes in wp-admin/includes/deprecated.php. */
+		'wp_user_search'                              => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_export_requests_table'       => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_removal_requests_table'      => 'wp-admin/includes/deprecated.php',
+	);
+
+	/**
+	 * Whether the autoloader has already been registered or not.
+	 *
+	 * Avoid registering the autoloader multiple times.
+	 *
+	 * @static
+	 * @access private
+	 *
+	 * @var bool
+	 */
+	private static $registered = false;
+
+	/**
+	 * Register the autoloader.
+	 *
+	 * @return void
+	 */
+	public static function register() {
+		// Bail early if already registered.
+		if ( self::$registered ) {
+			return;
+		}
+
+		self::register_external_bundled();
+		self::register_core();
+
+		self::$registered = true;
+	}
+
+	/**
+	 * Register the autoloader for external, bundled libraries.
+	 *
+	 * @return void
+	 */
+	public static function register_external_bundled() {
+		require_once ABSPATH . 'wp-includes/Requests/src/Autoload.php';
+		require_once ABSPATH . 'wp-includes/sodium_compat/autoload.php';
+
+		spl_autoload_register( array( '\WpOrg\Requests\Autoload', 'load' ) );
+	}
+
+	/**
+	 * Register the autoloader for WordPress Core classes.
+	 *
+	 * @return void
+	 */
+	public static function register_core() {
+		spl_autoload_register( array( __CLASS__, 'autoload_core' ), true, true );
+	}
+
+	/**
+	 * Autoload a WordPress class.
+	 *
+	 * @param string $class_name Class name.
+	 * @return void
+	 */
+	public static function autoload_core( string $class_name ): void {
```

Comment:
Return type 'void' is only allowed since PHP 7.1

```suggestion
	public static function autoload_core( string $class_name ) {
```

Reply:
Nice catch!

Reply:
Committed, got an error in the PHPCS checks so I reverted this change.
See https://github.com/WordPress/wordpress-develop/actions/runs/7814642791/job/21316365214?pr=3470
(`Error: void return type is not present in PHP version 7.0 or earlier`)

Reply:
@aristath: It's just the `void` that's the problem here (I had initially suggested `bool` a bit further up in the discussion, which is available in PHP 7.0, but then forgot that `void` is not when making the new suggestion).
The `string` for the method argument is fully supported and should be added, as in the referenced commits above.

Reply:
Done 👍